### ISSUE TITLE
[rewrite] invalid draw with stream

### ISSUE THREAD

**Issue by sixmen** [lhorie/mithril.js#1551]
### Description:

I used a stream for a checkbox. 'false->true' works, but 'true->false' does not work.

### Steps to Reproduce:

```
const value = stream(true);
const Component = {
  view: vnode => {
    return m('input[type="checkbox"]', {
      checked: value(),
      onclick: m.withAttr('checked', value)
    });
  }
}
m.mount(document.getElementById('app'), Component);
```

### Expected:

Click checkbox, and it will be unchecked.

### Actual:

It still remains as checked.

### Note:

Followings work well.

```
const state = {
  value: true,
  changeValue: v => {
    state.value = v;
  }
}
const Component = {
  view: vnode => {
    return m('input[type="checkbox"]', {
      checked: state.value,
      onclick: m.withAttr('checked', state.changeValue)
    });
  }
}
m.mount(document.getElementById('app'), Component);
```

or ```onclick: m.withAttr('checked', v => { value(v) })```

**Comment by gaspoute** [comment]
I think that the cause is ´m.withAttr('checked', value) is equivalent to ´m.withAttr('checked' v => value(v))´ (without parenthesis) then the listener returns the value of the stream. When the stream is ´false´, the listener returns ´false´ which is equivalent to ´e.preventDefault´. Correct me if I'm wrong :-)

**Comment by sixmen** [comment]
@gaspoute You're right. If I modify `m.withAttr`, it works well.

I used this pattern from 0.x which does not return result of callback.

master: https://github.com/lhorie/mithril.js/blob/5d2b6ea2b70df74fe215350f9884fec84a4c7071/mithril.js#L1504

rewrite: https://github.com/lhorie/mithril.js/blob/813bfb5643b7aad012da1d64575488ba682f454b/mithril.js#L1156

I think it must be fixed or documented as a breaking change.

**Comment by lhorie** [comment]
Fixed to match 0.2 behavior
